src/Compilers/CSharp/Test/Semantic/Semantics/UninitializedNonNullableFieldTests.cs

```diff
@@ -805,6 +979,69 @@ class C
             comp.VerifyDiagnostics();
         }
 
+        [Fact]
+        public void ValueType_NoConstructors()
+        {
+            var source =
+@"#nullable enable
+#pragma warning disable 0169
+struct S
+{
+    object F1;
+    static object F2;
+}";
+            var comp = CreateCompilation(source);
+            comp.VerifyDiagnostics(
+                // (6,19): warning CS8618: Non-nullable field 'F2' is uninitialized. Consider declaring the field as nullable.
+                //     static object F2;
+                Diagnostic(ErrorCode.WRN_UninitializedNonNullableField, "F2").WithArguments("field", "F2").WithLocation(6, 19));
+        }
+
+        [Fact]
+        public void ValueType_InstanceConstructor()
+        {
+            var source =
+@"#nullable enable
+#pragma warning disable 0169
+struct S
+{
+    object F1;
+    static object F2;
+    S(int i)
+    {
+        F1 = new object();
+    }
+}";
+            var comp = CreateCompilation(source);
+            comp.VerifyDiagnostics(
+                // (6,19): warning CS8618: Non-nullable field 'F2' is uninitialized. Consider declaring the field as nullable.
+                //     static object F2;
+                Diagnostic(ErrorCode.WRN_UninitializedNonNullableField, "F2").WithArguments("field", "F2").WithLocation(6, 19));
+        }
+
+        [Fact]
+        public void ValueType_StaticConstructor()
+        {
+            var source =
+@"#nullable enable
+#pragma warning disable 0169
+struct S
+{
+    object F1;
+    static object F2;
+    static object F3;
+    static S()
+    {
+        F2 = new object();
+    }
+}";
+            var comp = CreateCompilation(source);
+            comp.VerifyDiagnostics(
+                // (8,12): warning CS8618: Non-nullable field 'F3' is uninitialized. Consider declaring the field as nullable.
+                //     static S()
+                Diagnostic(ErrorCode.WRN_UninitializedNonNullableField, "S").WithArguments("field", "F3").WithLocation(8, 12));
+        }
```

Comment:
> } [](start = 7, length = 2)

consider testing in interface (state/field not allowed by DIM)

Reply:
What should we test given that fields are not allowed?

---
In reply to: [307937911](https://github.com/dotnet/roslyn/pull/37507#discussion_r307937911) [](ancestors = 307937911)